scripts/cleanup-azure-resources.sh

```
@@ -258,6 +288,44 @@ find_service_principals() {
+
+# Find resource groups by Azure tag
+find_resource_groups_by_tag() {
+    local tag_filter="$1"
+    local tag_key="${tag_filter%%=*}"
+    local tag_value="${tag_filter#*=}"
+
+    print_info "Searching for resource groups with tag '${tag_key}=${tag_value}'..." >&2
+
+    local rgs_json
+    rgs_json=$(az group list --tag "${tag_key}=${tag_value}" --query "[].{name: name, location: location}" -o json 2>/dev/null)
+    if [[ $? -ne 0 ]]; then
```

Shellcheck (0.11.0)

[style] Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.

(SC2181)